modules/cli/src/main/scala/scala/cli/commands/Fmt.scala

```diff
@@ -94,62 +37,39 @@ object Fmt extends ScalaCommand[FmtOptions] {
-    val (versionMaybe, confExists0) = readVersionFromFile(workspace, logger)
-    val cache                       = options.shared.buildOptions().archiveCache
-    val buildOptions                = options.buildOptions
-
-    val (version, confExists) = versionMaybe.map((_, confExists0)).getOrElse {
-      val errorMsg =
-        s"""|Scalafmt requires explicitly specified version.
-            |To configure the scalafmt version add the following line into .scalafmt.conf:
-            |    version = ${Constants.defaultScalafmtVersion} """.stripMargin
-      val msg = s"""|$errorMsg
-                    |Do you want to add Scalafmt version to .scalafmt.conf file?""".stripMargin
-      val scalafmtConfPath = workspace / ".scalafmt"
-      val entry = s"version = ${Constants.defaultScalafmtVersion}${System.lineSeparator()}"
-      InteractiveFileOps.appendToFile(interactive, msg, scalafmtConfPath, entry) { () =>
-        System.err.println(errorMsg)
-        sys.exit(1)
-      }
-      (Constants.defaultScalafmtVersion, true)
-    }
+    val (versionMaybe, dialectMaybe, pathMaybe) = readVersionAndDialectFromFile(workspace, logger)
+    val cache                                   = options.shared.buildOptions().archiveCache
+    val buildOptions                            = options.buildOptions
 
     if (sourceFiles.isEmpty)
       logger.debug("No source files, not formatting anything")
     else {
-
-      def scalaVerOpt = inputsOpt.flatMap { inputs =>
-        val (crossSources, _) =
-          CrossSources.forInputs(
-            inputs,
-            Sources.defaultPreprocessors(
-              buildOptions.scriptOptions.codeWrapper.getOrElse(CustomCodeWrapper),
-              buildOptions.archiveCache,
-              buildOptions.internal.javaClassNameVersionOpt
-            ),
-            logger
-          ).orExit(logger)
-        val sharedOptions = crossSources.sharedOptions(buildOptions)
-        sharedOptions
-          .scalaParams
-          .orExit(logger)
-          .map(_.scalaVersion)
-      }
-
-      def dialectOpt = options.dialect.map(_.trim).filter(_.nonEmpty).orElse {
-        scalaVerOpt.flatMap {
-          case v if v.startsWith("2.12.") => Some("Scala212")
-          case v if v.startsWith("2.13.") => Some("Scala213")
-          case v if v.startsWith("3.")    => Some("Scala3")
-          case _                          => None
+      val version =
+        options.scalafmtVersion.getOrElse(versionMaybe.getOrElse(Constants.defaultScalafmtVersion))
+      val dialectString = options.scalafmtDialect.getOrElse {
+        dialectMaybe.getOrElse {
+          version match
+            case v if v.startsWith("2.11.") => "scala211"
+            case v if v.startsWith("2.12.") => "scala212"
+            case v if v.startsWith("2.13.") => "scala213"
+            case v if v.startsWith("3.")    => "scala3"
+            case _                          => "default"
         }
       }
```

Comment:
NIT The nested `getOrElse` looks a bit ugly. This should do:
```suggestion
      val dialectString = options.scalafmtDialect.orElse(dialectMaybe).getOrElse {
        version match
          case v if v.startsWith("2.11.") => "scala211"
          case v if v.startsWith("2.12.") => "scala212"
          case v if v.startsWith("2.13.") => "scala213"
          case v if v.startsWith("3.")    => "scala3"
          case _                          => "default"
      }
```